test/AutoDiff/differentiable_attr_type_checking.swift

```diff
@@ -1,7 +1,22 @@
 // RUN: %target-swift-frontend -typecheck -verify %s
 
-@differentiable(vjp: dfoo) // expected-error {{'@differentiable' attribute cannot be applied to this declaration}}
-let x: Float = 1
+@differentiable // expected-error {{'@differentiable' attribute cannot be applied to this declaration}}
+let global: Float = 1
+
+func testLocalVariables() {
```

Comment:
Note: this PR didn't change the behavior of `@differentiable` on local variables. The `testLocalVariables` only documents the current behavior.

The way accessor names are displayed in diagnostics is not ideal and should be improved in a follow-up. Accessor names could show up like `the getter of 'getter' has no parameters ...` instead of `'_' has no parameters ...`.